### PR TITLE
Refactor graph editor and add more tests

### PR DESCRIPTION
**Pull request by thomaskrause**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] The *Unreleased* section in CHANGELOG.md has been amended to reflect the changes in this PR
- [X] Build (`mvn verify`) was run locally and any changes were pushed
- [X] The pull request is against the correct branch (`master` for bug fixes, `develop` for new functionality)

## Pull request type

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Static code analysis reported many issues in the `GraphEditor` class, including issues with readability because of too long and nested methods. This class alone contributed to a lot of the "Maintainability" score.
Also, the test coverage was very insufficient.

## What is the new behavior?

- Fixed all reported issues from SonarCloud in the `GraphEditor` class.
- Added more test cases. Because it is impossible to test  mouse wheels and we did the zoom exclusively with the mouse wheel, I added the <kbd>CTRL</kbd>+<kbd>+</kbd> <kbd>CTRL</kbd>+<kbd>-</kbd> shortcuts which also perform zoom. This new feature is documented, everything else is just refactoring

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Does this introduce new dependencies?

- [ ] Yes
- [X] No

If this introduces new dependencies:

- [ ] [Dependencies and citation templates](https://github.com/hexatomic/hexatomic/tree/develop/releng/templates) have been updated where necessary

## Other information

---

# Checklist for maintainers

## Releases

- [ ] Do **NOT** push the green merge button on GitHub.  
Follow the latest version of the [developer/maintainer docs](https://hexatomic.github.io/hexatomic/dev/) for making releases.
- [ ] License and citation information for dependencies are complete (each dependency has a folder in `/THIRD-PARTY/`).

### CODE REVIEW

File: bundles/org.corpus_tools.hexatomic.graph/src/main/java/org/corpus_tools/hexatomic/graph/GraphEditor.java [hunk 1]
```diff
@@ -443,136 +329,163 @@ private void updateView(final boolean recalculateSegments, final boolean scrollT
...
       final List<SegmentSelectionEntry> oldSelectedSegments = new LinkedList<>();
+      final List<SegmentSelectionEntry> newSelectedSegments = new LinkedList<>();
 
       if (recalculateSegments) {
         // Store the old segment selection
         for (TableItem item : textRangeTable.getSelection()) {
           SegmentSelectionEntry entry = new SegmentSelectionEntry();
-          entry.range = (Range<Long>) item.getData("range");
+          entry.range = (Range<Long>) item.getData(RANGE);
           entry.text = (STextualDS) item.getData("text");
```
Comment (sdruskat) [inline bundles/org.corpus_tools.hexatomic.graph/src/main/java/org/corpus_tools/hexatomic/graph/GraphEditor.java:341]: Shouldn't this also be moved into a constant?

File: docs/user/src/usage/graph-editor/README.md [hunk 2]
```diff
@@ -27,8 +27,10 @@ The left-hand side of the graph editor is taken up by the graph view, which disp
 
 You can navigate the graph view as follows:
 
-- **Zoom in and out** by using the mouse wheel.
+- **Zoom in and out** by using the **mouse wheel**.
   - You will zoom in to where your mouse cursor is.
+- **Zoom in and out** by using **using the keyboard**.
+  - If you press and hold the <kbd>Ctrl</kbd> key you can zoom in with the <kbd>+</kbd> key and zoom out with <kbd>-</kbd> key.
```
Comment (sdruskat) [inline docs/user/src/usage/graph-editor/README.md:33]:
```suggestion
  - If you press and hold the <kbd>Ctrl</kbd> key, you can zoom in with the <kbd>+</kbd> key and zoom out with the <kbd>-</kbd> key.
```

Insert comma for readability, and add missing "the".

File: tests/org.corpus_tools.hexatomic.it.tests/src/main/java/org/corpus_tools/hexatomic/it/tests/TestGraphEditor.java [hunk 3]
```diff
@@ -170,16 +243,91 @@ public String getFailureMessage() {
...
+
+    // Initially, the zoom is adjusted to match the height, so moving up/down should not do anything
+    keyboard.pressShortcut(Keystrokes.DOWN);
+    bot.waitUntil(
+        new ViewLocationReachedCondition(viewPort, new Point(origLocation.x, origLocation.y)));
+    keyboard.pressShortcut(Keystrokes.UP);
+    bot.waitUntil(
+        new ViewLocationReachedCondition(viewPort, new Point(origLocation.x, origLocation.y)));
+
+    // Zoom in so we can move the view with the cursors
```
Comment (sdruskat) [inline tests/org.corpus_tools.hexatomic.it.tests/src/main/java/org/corpus_tools/hexatomic/it/tests/TestGraphEditor.java:276]:
```suggestion
    // Zoom in so we can move the view with the arrow keys
```

Sorry, slightly pedantic.

File: bundles/org.corpus_tools.hexatomic.graph/src/main/java/org/corpus_tools/hexatomic/graph/GraphEditor.java (hunk 1 above)
Comment (thomaskrause) [reply bundles/org.corpus_tools.hexatomic.graph/src/main/java/org/corpus_tools/hexatomic/graph/GraphEditor.java]: I "text" into a constant but "range" was already in one.

**Pull request merged**